src/main/java/com/p6spy/engine/spy/P6DataSource.java

```diff
@@ -182,14 +182,14 @@ private void setDataSourceProperties(HashMap<String, String> props) throws SQLEx
               // expects
               String value = props.get(key);
               Class<?>[] types = method.getParameterTypes();
-              if (types[0].getName().equals(value.getClass().getName())) {
+              if (types[0].isAssignableFrom(value.getClass())) {
```

Comment:
in line 183 `value` is of type `String`, so I guess `value.getClass()` can be replaced with `String.class`